### PR TITLE
DOC: whatsnew fixups

### PR DESCRIPTION
**Pull request by TomAugspurger**
* Refer to public API
* Fixed formatting
* Remove ref to undocumented function

### CODE REVIEW

File: doc/source/whatsnew/v1.0.0.rst [hunk 1]
```diff
@@ -574,7 +574,7 @@ or ``matplotlib.Axes.plot``. See :ref:`plotting.formatters` for more.
 - :meth:`Series.where` with ``Categorical`` dtype (or :meth:`DataFrame.where` with ``Categorical`` column) no longer allows setting new categories (:issue:`24114`)
 - :class:`DatetimeIndex`, :class:`TimedeltaIndex`, and :class:`PeriodIndex` constructors no longer allow ``start``, ``end``, and ``periods`` keywords, use :func:`date_range`, :func:`timedelta_range`, and :func:`period_range` instead (:issue:`23919`)
 - :class:`DatetimeIndex` and :class:`TimedeltaIndex` constructors no longer have a ``verify_integrity`` keyword argument (:issue:`23919`)
-- :func:`core.internals.blocks.make_block` no longer accepts the "fastpath" keyword(:issue:`19265`)
```
Comment (jbrockmendel) [inline doc/source/whatsnew/v1.0.0.rst]: is `:func:` not a thing?  im pretty sure this one was my doing

File: doc/source/whatsnew/v1.0.0.rst [hunk 2]
```diff
@@ -574,7 +574,7 @@ or ``matplotlib.Axes.plot``. See :ref:`plotting.formatters` for more.
...
+- ``pandas.core.internals.blocks.make_block`` no longer accepts the "fastpath" keyword(:issue:`19265`)
```
Comment (jbrockmendel) [inline doc/source/whatsnew/v1.0.0.rst]: missing space after "keyword"

Comment (jbrockmendel) [reply doc/source/whatsnew/v1.0.0.rst]: same in the line below (also my fault)

File: doc/source/whatsnew/v1.0.0.rst (hunk 1 above)
Comment (TomAugspurger) [reply doc/source/whatsnew/v1.0.0.rst]: It is, but we don't have an autodoc entry for `core.internals.blocks.make_block`, so the link isn't created.

**Comment by jbrockmendel** [comment]
Thanks @TomAugspurger

**Pull request merged**